src/sql-parser/src/parser.rs

```diff
@@ -1522,6 +1533,11 @@ impl<'a> Parser<'a> {
         } else {
             Default::default()
         };
+        let compression = if self.parse_keyword(COMPRESSION) {
+            self.parse_compression()?
+        } else {
+            Default::default()
+        };
```

Comment:
I spoke with @benesch  about the syntax. We don't need to make this a WITH option, but we can parse it as part of the connector (in `self.parse_connector` above) but only for s3 and file connectors. Or, if you prefer, we could also make it a `WITH` option - Nikhil didn't have a strong opinion.

As you mentioned in the meeting today, indeed, the current syntax makes it seem like gzip compression is an orthogonal component which could in principle apply to many source types, which (IMO) isn't likely to be the case.